### ISSUE TITLE
GameFinder stops looking for Steam games if any Steam library isn't available

### ISSUE THREAD

**Issue by Frogperson** [erri120/GameFinder#133]
I mistakenly reported this issue to Wabbajack https://github.com/wabbajack-tools/wabbajack/issues/2518

As a quick recap, I have multiple Steam libraries on different drives. If one of these drives isn't plugged in, then GameFinder completely errors out rather than continuing to check my other libraries.

Dynamically removable libraries are an officially supported feature in Steam, so I'm hoping GameFinder can update to properly support them.

**Comment by Nutzzz** [comment]
You'll note that the exception you report actually occurs in *another* library that in turn GameFinder is using, [NexusMods.Paths](https://github.com/Nexus-Mods/NexusMods.Paths), though GameFinder's author @erri120 has had a hand in it.

In any case, I was eventually able to reproduce this.  With both drive letters and path mounts located in a drive's root directory, no exception occurred for me with both 4.2.0 and 4.1.0 (the version Wabbajack 3.5.0.1 uses) after a drive is removed; there's an error message that the path isn't found, then it continues to search for additional locations.  Steam wouldn't let me create a library with a path mount more than one directory deep off the root (it reports that it's not "empty" even if it is).  I had to create the library first, move the data to the new drive, then create a path mount in the folder it was expecting.

The exception occurs on line 90 of SteamHandler.cs in libraryFolder.EnumerateAppManifestFilePaths().  Though there is a check that the directory exists, there isn't a check whether the "steamapps" subdir exists before it attempts to enumerate.

I've got a PR incoming...
